src/components/MovieListPage.js

```diff
@@ -1,23 +1,43 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import Search from './Search';
 import MovieListItem from './MovieListItem';
+import Spinner from './Spinner';
+import API_ROOT_URL from '../constants';
+import ErrorNotification from './ErrorNotification';
 
 class MovieListPage extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      movies: [],
       filterText: '',
+      error: false,
+      isLoading: true,
     };
   }
 
-  handleFilterTextChange = (value) => {
-    this.setState({ filterText: value });
+  componentDidMount() {
+    this.fetchMovies();
+  }
+
+  fetchMovies = async () => {
+    const moviesApiUrl = `${API_ROOT_URL}`;
+
+    try {
+      const res = await fetch(moviesApiUrl);
+      if (!res.ok) {
+        this.setState({ error: true, isLoading: false });
+        throw new Error('Incorrect response');
+      }
+      const data = await res.json();
+      this.setState({ movies: data, isLoading: false });
+    } catch (e) {
+      console.log('An error occurred:', e);
+    }
   };
```

Comment:
One interesting exercise would be to extract the `fetchMovies` and `fetchMovie` above to a separate `api.js` file and see how the code changes.

Reply:
Done, looks cleaner from the component perspective, but I am not sure if the solution is optimal (`getApiData` is returning object which is then used in component to manage state). Let me know what you think.